### PR TITLE
Add util functions for tokens

### PR DESCRIPTION
**Pull request by lobkovilya**
## Description
Add two functions:
- `GenerateToken(ctx context.Context, p Provider, expiresAt time.Duration) (string, error)`
- `VerifyToken(token string, x509crt *x509.Certificate) error` 

Cover them with unit tests.

Signed-off-by: Ilya Lobkov <ilya.lobkov@xored.com>

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -201,3 +201,6 @@ issues:
       linters:
         - dupl
       text: "lines are duplicate of"
+    - path: pkg/tools/security/token_test.go
+      linters:
+        - gochecknoinits
```
Comment (denis-tingajkin) [inline .golangci.yml]: Why do we need to disable this linter?

File: pkg/tools/security/token_test.go [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+	testCA, err = generateCA()
+	if err != nil {
+		panic(err)
+	}
+
+	testTLSCertificate, err = generateKeyPair(spiffeID, "test.com", &testCA)
+	if err != nil {
+		panic(err)
+	}
+}
```
Comment (denis-tingajkin) [inline pkg/tools/security/token_test.go]: Is it possible to use testify/suite for setup tests? Tesitfy suites can have setup/teardown methods. take a look at https://github.com/stretchr/testify/blob/master/suite/interfaces.go#L20

File: .golangci.yml (hunk 1 above)
Comment (lobkovilya) [reply .golangci.yml]: Fixed

File: pkg/tools/security/token_test.go (hunk 2 above)
Comment (lobkovilya) [reply pkg/tools/security/token_test.go]: Fixed

**Pull request merged**